docs/source/reference/trial.rst

```diff
@@ -1,6 +1,8 @@
 optuna.trial
 ============
 
+The :mod:`optuna.trial` module contains :class:`~optuna.trial.Trial` related classes and functions. A :class:`~optuna.trial.Trial` instance represents a process of evaluating an objective function. This instance is passed to an objective function and provides interfaces to get parameter suggestion, manage the trial's state, and set/get user-defined attributes of the trial, so that optuna users can define a custome objective function through the interfaces. Basically, optuna users only treats it in their custom objective functions.
```

Comment:
In addition to @toshihikoyanase's comment, I have several minor suggestions.
- A nit: :mod:`optuna.trial` -> :mod:`~optuna.trial`
- The second sentence onwards is an explanation of `Trial`, which has nothing to do with `FrozenTrial` etc. How about breaking it up into paragraphs?
```suggestion
The :mod:`~optuna.trial` module contains :class:`~optuna.trial.Trial` related classes and functions. 

A :class:`~optuna.trial.Trial` instance represents a process of evaluating an objective function. This instance is passed to an objective function and provides interfaces to get parameter suggestion, manage the trial's state, and set/get user-defined attributes of the trial, so that Optuna users can define a custom objective function through the interfaces. Basically, Optuna users only use it in their custom objective functions.
```